File: C#/Binder/ContentBinder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using XmlBinder.Interface;
using XmlBinder.Tools;

namespace XmlBinder.Binder
{
	/// <summary>
	/// Binder to bind the content of Xml tag.
	/// </summary>
	public class ContentBinder : XmlBinder.Abstract.AbstractBinder
	{
		/// <summary>
		/// Transformer to apply before saving data from xml.
		/// </summary>
		protected TransformerInterface m_transformer;

		/// <summary>
		/// Transformer to apply on data.
		/// </summary>
		public TransformerInterface Transformer
		{
			get
			{
				return m_transformer;
			}
		}

		/// <summary>
		/// Create a content binder to bind data turn with transformer into the field name.
		/// </summary>
		/// <param name="_fieldName">The name of the field to save data.</param>
		/// <param name="_transformer">The transformer to use to turn data into other format.</param>
		public ContentBinder(string _fieldName, TransformerInterface _transformer) : base(_fieldName)
		{
			m_transformer = _transformer;
		}

		public override object read(object _source, string _content)
		{
			object data = Transformer.read(_content);
			FieldAccess.setField(_source, FieldName, data);
			return data;
		}

		public override string write(object _source)
		{
			//object data = FieldAccess.getField(_source, FieldName);
			//return Transformer.write(data);
			return Transformer.write(_source);
		}
	}
}

File: C++/XmlBinder/Binder/contentbinder.cpp
#include "contentbinder.hpp"
#include "Tools/fieldaccess.hpp"

ContentBinder::ContentBinder(const QString & _fieldName, ITransformer * _transformer)
	: AbstractBinder(_fieldName)
{
	m_transformer = _transformer;
}

QObject * ContentBinder::read(QObject * _source, QString _content)
{
	QVariant value = m_transformer->read(_content);
	FieldAccess::setValue(_source, this->fieldName(), value);
	return _source;
}

QPair<QString, QObject *> ContentBinder::write(QObject * _source)
{
	QVariant v = FieldAccess::value(_source, this->fieldName());
	return QPair<QString, QObject*>(m_transformer->write(v), _source);
}
